### PR TITLE
Fix most SA errors, remove Trace feature, fix Header usage

### PR DESCRIPTION
**Pull request by DanielBadura**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
     "symfony/asset-mapper": "^7.0",
     "symfony/config": "^5.4.31|^6.4.0|^7.0.0",
     "symfony/dependency-injection": "^5.4.33|^6.4.1|^7.0.1",
+    "symfony/expression-language": "^7.2",
```
Comment (DanielBadura) [inline composer.json]: need to adjust version like the others

File: src/Controller/EventController.php [hunk 2]
```diff
@@ -57,6 +61,9 @@ private function listenerMethods(string $eventClass): array|null
         );
```
Comment (DanielBadura) [inline src/Controller/EventController.php:65]: `$this->listenerProvider->listenersForEvent($eventClass)` return `iterable`, needs to be casted to `array` here

File: src/DependencyInjection/CustomMessageHeaderCompilerPass.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+namespace Patchlevel\EventSourcingAdminBundle\DependencyInjection;
+
+use Patchlevel\EventSourcing\Metadata\Message\MessageHeaderRegistry;
+use Patchlevel\EventSourcingAdminBundle\Controller\DefaultController;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/** @interal */
+final class CustomMessageHeaderCompilerPass implements CompilerPassInterface
```
Comment (DavidBadura) [inline src/DependencyInjection/CustomMessageHeaderCompilerPass.php:13]: I'm not entirely happy that we have to do it this way, but there's no other option at the moment. So it's fine for now. But I'd like to see at some point whether we can add something to the event sourcing lib to make it work better.

**Review by DavidBadura — APPROVED** [review]

File: src/DependencyInjection/CustomMessageHeaderCompilerPass.php (hunk 3 above)
Comment (DanielBadura) [reply src/DependencyInjection/CustomMessageHeaderCompilerPass.php:13]: Yeah, imho its related to this issue here: https://github.com/patchlevel/event-sourcing/issues/524

**Comment by DanielBadura** [comment]
@DavidBadura I added the `iterator_to_array` and rebased/squashed some commits. This is now ready from my side.

File: tests/Unit/PatchlevelEventSourcingBundleTest.php [hunk 4]
```diff
@@ -46,21 +55,41 @@ public function testEnabled(): void
...
+            [
+                'patchlevel_event_sourcing' => [
+                    'connection' => [
+                        'url' => 'sqlite3:///:memory:',
+                    ],
+                ]
+            ],
+            $container
+        );
+
```
Comment (DanielBadura) [inline tests/Unit/PatchlevelEventSourcingBundleTest.php:92]: I dislike this part here. I'm not sure though how to solve this in a better way.

**Pull request merged**